### PR TITLE
READY flash fixes

### LINKED ISSUE #36098: assemblies made with flashers dont work
no way to activate the flash

### PR DESCRIPTION
**Pull request by PKPenguin321**
:cl:
fix: Restored accidentally cut functionality to flashes. You can once again attach them to a signaller and signal them to trigger an AoE flash remotely.
/:cl:

Fixes https://github.com/tgstation/tgstation/issues/36098

Fixes a message that would incorrectly output "Your the flash emits a blinding light!"

### CODE REVIEW

File: code/modules/assembly/flash.dm [hunk 1]
```diff
@@ -81,6 +81,8 @@
 /obj/item/device/assembly/flash/proc/get_flash_targets(atom/target_loc, range = 3, override_vision_checks = FALSE)
 	if(!target_loc)
 		target_loc = loc
+		if(!isturf(target_loc))
+			target_loc = loc.loc
```
Comment (PKPenguin321) [inline code/modules/assembly/flash.dm]: so the flash is actually contained in an assembly holder object meaning its loc is not a turf or something and as far as i can tell it breaks it. the AOE flash happens but if youre standing next to it you're not affected by it. I THINK this is the part of it that is broken. and what i've added here didn't seem to fix it, either.

**Comment by ExcessiveUseOfCobblestone** [comment]
https://github.com/PKPenguin321/-tg-station/blob/59b235f24c2de99d68ee9aefdbeaf4bf2efb72b2/code/modules/assembly/flash.dm#L74

just uh try removing "mob" and see if that fixes anything lol.

File: code/modules/assembly/flash.dm (hunk 1 above)
Comment (Supermichael777) [reply code/modules/assembly/flash.dm]: use get_turf

**Comment by ExcessiveUseOfCobblestone** [comment]
ok but for real is `/list/mob/targets` a thing and what's the benefit of doing that?

**Comment by PKPenguin321** [comment]
this isnt WIP anymore

File: code/modules/assembly/flash.dm [hunk 2]
```diff
@@ -159,6 +159,9 @@
 	burn_out()
 	. = ..()
 
+/obj/item/device/assembly/flash/activate()
+	AOE_flash()
```
Comment (ninjanomnom) [inline code/modules/assembly/flash.dm]: The parent has a use cooldown check, just add
```
if(!..())
  return
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by PKPenguin321** [comment]
review addressed

**Review by ninjanomnom — APPROVED** [review]

**Pull request merged**